shared/StringCommon.F90

```diff
@@ -0,0 +1,219 @@
+      do i = 1, size(array)
+         string(i:i) = array(i)
+      end do
+
+   end function to_string
+
+   function to_character_array(s) result(ch)
+      character, allocatable :: ch(:)
+      character(len=*), intent(in) :: s
+      integer :: i
+
+      allocate(ch(len(s)))
+      do i= 1, size(ch)
+         ch(i) = s(i:i)
+      end do
+
+   end function to_character_array
+
+   !===============================================================================
+   ! Inquiry functions - character
+   !===============================================================================
+
+   elemental logical function is_lower(ch)
+      character, intent(in) :: ch
+
+      is_lower = ch >= 'a' .and. ch <= 'z'
+
+   end function is_lower
+
+   elemental logical function is_upper(ch)
+      character, intent(in) :: ch
+
+      is_upper = ch >= 'A' .and. ch <= 'Z'
+
+   end function is_upper
+
+   elemental logical function is_alpha(ch)
+      character, intent(in) :: ch
+
+      is_alpha = is_lower(ch) .or. is_upper(ch)
+
+   end function is_alpha
+
+   elemental logical function is_digit(ch)
+      character, intent(in) :: ch
+
+      is_digit = (index(DIGITS, ch) > 0)
+
+   end function is_digit
+
+   elemental logical function is_alphanum_character(ch, exclude_underscore)
+      character, intent(in) :: ch
+      logical, optional, intent(in) :: exclude_underscore
+      logical :: incl_und_
+
+      incl_und_ = .TRUE.
+      if(present(exclude_underscore)) incl_und_ = .not. exclude_underscore
+      is_alphanum_character = is_alpha(ch) .or. is_digit(ch) .or. (incl_und_ .and. ch == '_')
+
+   end function is_alphanum_character
+
+   !===============================================================================
+   ! Inquiry functions - string (character with any length)
+   !===============================================================================
+
+   logical function is_alpha_only(s)
+      character(len=*), intent(in) :: s
+
+      is_alpha_only = len(s) > 0 .and. all(is_alpha(to_character_array(s)))
+
+   end function is_alpha_only
+
+   logical function is_numeric(s)
+      character(len=*), intent(in) :: s
+
+      is_numeric = len(s) > 0 .and. all(is_digit(to_character_array(s)))
+
+   end function is_numeric
+
+   logical function is_alphanumeric(s, exclude_underscore)
+      character(len=*), intent(in) :: s
+      logical, optional, intent(in) :: exclude_underscore
+      logical :: exclude_underscore_
+
+      exclude_underscore_ = .FALSE.
+      if(present(exclude_underscore)) exclude_underscore_ = exclude_underscore
+      is_alphanumeric = len(s) > 0 .and. all(is_alphanum_character(to_character_array(s), exclude_underscore_))
+
+   end function is_alphanumeric
+
+   !===============================================================================
+   ! Character conversion utilities
+   !===============================================================================
+
+   elemental function lowercase(ch) result(th)
```

Comment:
I don't think elemental works the way we would want here.   We don't tend to have arrays of characters.  strings are different.

Reply:
The idea was to avoid iterating over the characters in a string to apply a function to all of them. If it is better to iterate over the characters in the string, I will remove and add the iteration wherever a function acts on all the characters.